---
title: "Akkermansia muciniphila"
author: Jessica Hunter
date: May 10, 2022
output:
  prettydoc::html_pretty:
    theme: hpstr
    highlight: github
---

```{r setup, include=FALSE}
library(flexdashboard)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, error = FALSE)
```




```{r, results='hide', include=FALSE}
library(tidyverse)
library(here)
library(janitor)
library(wordcloud)
library(reshape2)
library(magick)
```

```{r include=FALSE}
###### reading in files

raw16s <- read_csv(here("finalproject", "data", "CDI_Samples_16S.csv"))  
rawq1 <- read_csv(here("finalproject","data","questg1.csv")) 
rawq2 <- read_csv(here("finalproject","data","questg2.csv"))
rawq4 <- read_csv(here("finalproject","data","questg4.csv"))
rawq0 <- read_csv(here("finalproject","data","questg0.csv"))

allquest<- rawq1 %>% 
  full_join(rawq2) %>% 
  full_join(rawq4) %>% 
  full_join(rawq0) %>% 
  clean_names() %>% 
  select(!22:52) %>% 
  mutate(presence = if_else(phylogroup=="Am0", "Not present", "Present"))

rawdiet <- read_csv((here("finalproject", "data", "dietdata.csv"))) %>% 
  clean_names()

###### reading in picture image

amucem<- image_read(here("finalproject", "data", "EMamuc.png")) 
amucobesity<- image_read(here("finalproject","data", "amucobesity.jpg"))
mpex <- image_read(here("finalproject", "data", "multiplexexample.png")) 
control <- image_read(here("finalproject", "data", "control.jpg")) 
results <- image_read(here("finalproject", "data", "results.jpg")) 
zoomout <- image_read(here("finalproject", "data", "zoomedout.jpg"))
zoom <- image_read(here("finalproject", "data", "zoom.jpg")) 
happy <- image_read(here("finalproject", "data", "happygut.png"))
```

# Background

-----

### *Akkermansia muciniphila*


-----
```{r echo=FALSE}
amucem
```
---

* Akkermansia muciniphila is a mucin-loving microbe that usuaally represents 1 to 5%  of our intestinal microbes. It's mucin preference thickens our gut barrier.

* It improves glucose intolerance and insulin resistance to inprove diabetic symptoms. 

* It reduces high-fat-diet weight gain and reduces inflammation.

----

```{r echo=FALSE}
amucobesity
```
---

## Data source

---


### Study Details regarding Sample Aquisition

#### "Dietary Habits are associated with Gut Composition"

[https://doi.org/10.1016/j.jand.2019.07.024](https://doi.org/10.1016/j.jand.2019.07.024)  

Samples came from:   
- 130 children ages 3-9  
- from suburban Los Angeles  
- Between June and September 2014  

---

## My Project:

#### "Designing a multiplex PCR assay to detect the different strains of *A. muciniphila*"

```{r echo=FALSE}
mpex
```

- 16S sequencing doesn't distinguish strain types because too similar
- cheaper, faster, than isolation and whole genome sequencing 
- Designed primers to amplify exclusive sequences of different lengths from each strain

---

## Results

#### Testing with controls

```{r echo=FALSE}
control
```

---

#### Some of the actual results

```{r echo=FALSE}
results
```

---

##### More actual results
```{r echo=FALSE}
zoomout
```

---

#### This person has 2 strains

```{r echo=FALSE}
zoom 
```

---
 
```{r, include=FALSE}
###### cleaning data and sorting
clean16s2 <- raw16s %>% # reading in file
  clean_names() %>% # converting the column names to easier format
   select(sample_id = taxon, # renaming columns and selecting column I want
         amuc16s = akkermansia_ra_16s, # renaming columns
         am_i, am_ii, am_iv, # selecting column I want
         dna_conc = dna_concentration) %>% # renaming columns
  mutate(am_i, am_i = if_else(am_i == "+++", 4, # changing symbols to numbers
                                 if_else(am_i == "++", 3,
                                 if_else(am_i == "+", 2,
                                 if_else(am_i == "+/-", 1,
                                 if_else(am_i == "-", 0, 0)))))) %>% 
  mutate(am_ii, am_ii = if_else(am_ii == "+++", 4, # changing symbols to numbers
                                 if_else(am_ii == "++", 3,
                                 if_else(am_ii == "+", 2,
                                 if_else(am_ii == "+/-", 1,
                                 if_else(am_ii == "-", 0, 0)))))) %>% 
  mutate(am_iv, am_iv = if_else(am_iv == "+++", 4, # changing symbols to numbers
                                 if_else(am_iv == "++", 3,
                                 if_else(am_iv == "+", 2,
                                 if_else(am_iv == "+/-", 1,
                                 if_else(am_iv == "-", 0, 0)))))) %>% 
  mutate(ampercent = str_replace_all(amuc16s,"%", ""), # removes the % symbol
         dna_conc = as.numeric(dna_conc), # converts to numeric
         ampercent = as.numeric(ampercent)) %>%  # converts to numeric
  drop_na() %>%  # removes na's which removes the non akkermansia
  mutate(totaldna = dna_conc * ampercent/100) # creating a column with calculated total DNA

########## turning data into long data

clean16slong <- clean16s2 %>% # creating new long dataset
  pivot_longer(  # changes from wide to long data
    cols = am_i:am_iv, # select columns
    names_to = "Strain", # name of column of combined variables
    values_to = "value") %>% # new column
  filter(value != 0) %>% # removes the ones that aren't strains
  mutate( # change values in column
    Strain = if_else(Strain == "am_i", # if condition
    "AmI", # if true
    if_else(Strain == "am_ii", # next condition if not true
    "AmII", # if true
    "AmIV"))) # if false
    
```


---

# Data Visualizations 

### Total DNA needed to detect A. muciniphila

```{r echo=FALSE}
ggplot()+
  geom_jitter(clean16slong, # creates a jitter plot with this data
              mapping = aes(x = value, # maps x to plot
                            y = totaldna, # maps y to plot
                            color = Strain), # colors based on strain
              width = 0.25)+ # width of points
  labs(x = "PCR detection band strength", 
       y = "DNA in ng") 
```



----------

This shows the amount of nanograms of geonomic DNA that were present for detection. 

As the amount of DNA is increased the band strength is stronger

------------

### This is the relative percent of A.muciniphila present for each of the samples out of the total gut community 

```{r echo=FALSE}
ggplot(clean16slong, mapping = aes(x = Strain, # map x to plot
                                   y = ampercent, # map y to plot
                                   color = Strain))+
  geom_jitter()+ # creates a jitter plot
  labs(y = "Percent relative abundance") # labels y axis
```





------------

# Nutritional

```{r, include=FALSE}
### cleaning data ###

newdietdata <- rawdiet %>% # create a new dataset
  full_join(allquest) %>% # adding the phylogroups
  select(!68:105) %>%  # remove colums I don't need
  mutate(idnumber = as.numeric(str_replace_all(personal_id, "[^[:digit:]]", ""))) %>% # remove all of the letters
  drop_na() %>% # remove the rows with missing data
  rename(carbpercent = percent_kcal_from_carbohydrate, # rename the columns
         fatpercent = percent_kcal_from_fat,
         proteinpercent = percent_kcal_from_protein) 

###### creating long data from wide data

nutrientlong <- newdietdata %>% # creating a long new dataset
  select(carbpercent,proteinpercent, fatpercent, phylogroup) %>% # selecting the columns I want
  pivot_longer( # convert the data from wide to long
    cols = carbpercent:fatpercent, # selecting the columns to convert
    names_to= "type", # name of new column
    values_to = "percentvalue") # name of new column with values

meatlong <- newdietdata %>% # creating a long new dataset
  pivot_longer( # convert the data from wide to long
    cols = meat:seafood_low_fa,  # selecting the columns to convert
    names_to = "MeatType") %>% # name of new column with combined data
  select(phylogroup, MeatType, value) %>% # selecting the columns I want
  filter(value != 0) # remove the values with no value

```




### Nutritional Breakdown of meals consumed within 24 hours of sample being taken

```{r echo=FALSE}
ggplot(nutrientlong, # data used
       aes(x= phylogroup, # map x
           y=percentvalue, # map y
           color = type))+ # colors determined by nutrient type
  geom_point( # creates a point plot with data
    position = position_jitterdodge())+ # spreading out the points the points
  labs(x = "Type of Akkermansia", 
       y = "Percent of Nutrients Consumed")+ # y title
   scale_color_discrete(name = "Nutrient Type",
                      labels = c("Carbohydrates", "Fat", "Protein"))+
  scale_x_discrete(labels = c("None", "AmI", "AmII", "AmIV"))
```


---------

There are slight diet differences among the different strains and those that have none

-------

```{r echo=FALSE}

ggplot(meatlong, # data used
       aes(x= phylogroup, # mapping x to plot
           y = value, # mapping y to plot
           color = MeatType))+ # colored by type of meat
  geom_point(position = position_jitterdodge()) + # spread the points out
  labs(title = "Grams of meat consumed by each type",
       x = "Akkermansia Type", # labels x axis
       y = "grams",) + # labels y axis
  facet_wrap(~MeatType) + # facets the plots
  scale_color_discrete(name = "Meat Type",
                      labels = c("Red Meat", "Poultry", "Processed Meats", "High Fatty Acid Seafood", "Low Fatty Acid Seafood"))+
  scale_x_discrete(labels = c("None", "AmI", "AmII", "AmIV"))
```

----

#### These graphs show some of the demographic distributions of the households that have Akkermansia and those that don't



```{r echo=FALSE}
 
 ggplot(allquest, mapping = aes(x= presence, y = wks_preg, fill = presence))+
   geom_violin()+
  labs(x = "Presence of Akkermansia",
       y = "Number of weeks pregnant",
       title = "Pregnancy term distribution of test subjects")

 ggplot(allquest, mapping = aes(x= presence, y = birth_wt_g, fill = presence))+
   geom_violin()+
  labs(x = "Presence of Akkermansia",
       y = "Birth weight in grams",
       title = "Birth weight distribution of test subjects")

 ggplot(allquest, mapping = aes(x= presence, y = weaned_real_months, fill = presence))+
   geom_violin()+
  labs(x = "Presence of Akkermansia",
       y = "Number of months old when weaned",
       title = "Age distribution of test subjects when weaned")
 
ggplot(allquest, aes(x = age_years, fill = presence, color = presence))+
  geom_dotplot(alpha = 0.6, binwidth = .42)+
  labs(x = "years in age",
       y = "count",
       title = "Age distribution of test subjects")

ggplot(allquest, aes(x = str_wrap(ethnicity, width = 35), fill = presence))+
  geom_dotplot(alpha = 0.5,binwidth = .45)+
  labs(x = "Ethnicity",
       y = "Count",
       title = "Ethnicity distribution of test subjects")+
  coord_flip()+
  theme(legend.position = c(0.8,0.2))

ggplot(allquest, mapping = aes(x= presence, y = max_income, fill = presence))+
   geom_violin()+
  labs(x = "Presence of Akkermansia",
       y = "Maximun Household Income in USD",
       title = "Income distribution of test subjects")

```

--- 

#### This shows the foods and vitamins associated with those that have *Akkermansia* and those that don't


```{r, echo=FALSE}

dietwords <- newdietdata %>% # creates new dataset by editing
  mutate(presence = if_else(# creates a new column
    phylogroup == "Am0", # condition
    "No Amuc", # True
    "Amuc Present")) %>% # if false print this
  select(!c(kcal,carbpercent, proteinpercent, fatpercent,  dairy_total,idnumber, phylogroup, 1:3)) %>% # removing columns I don't want
  group_by(presence) %>% # group by present of not
  summarise_if(is.numeric, mean, na.rm = TRUE) %>% 
   pivot_longer(cols= !presence, names_to = "type") %>% # making the data back long for my plot
    filter(value!=0) %>%  # removing the empty values
  drop_na() %>% # remove rows with missing data
  acast(type~presence, value.var = "value", fill = 0) %>% # creates dataset
  comparison.cloud( # compares data by values and plots in wordcloud
    colors = c("darkorange", "blue"), # colors for my word cloud
    max.words = 100, # maximum amount
    random.order = FALSE, # in decreasing order by value
    title.size = 2, # size of title
    rot.per = .3, # percent rotated
    scale = c(4, 1)) # large to small scale

```

----

# References

1. Plovier H, Everard A, Druart C, Depommier C, Van Hul M, Geurts L, Chilloux J, Ottman N, Duparc T, Lichtenstein L, Myridakis A, Delzenne NM, Klievink J, Bhattacharjee A, van der Ark KCH, Aalvink S, Martinez LO, Dumas M-E, Maiter D, Loumaye A, Hermans MP, Thissen J-P, Belzer C, de Vos WM, Cani PD. 2017. A purified membrane protein from Akkermansia muciniphila or the pasteurized bacterium improves metabolism in obese and diabetic mice. Nat Med 23:107–113.
2. Why Gut Health Matters. Lifeway Kefir. https://lifewaykefir.com/why-gut-health-matters/. 
3. Herman DR, Rhoades N, Mercado J, Argueta P, Lopez U, Flores GE. 2020. Dietary Habits of 2- to 9-Year-Old American Children Are Associated with Gut Microbiome Composition. Journal of the Academy of Nutrition and Dietetics 120:517–534.
4. Cani PD, de Vos WM. 2017. Next-Generation Beneficial Microbes: The Case of Akkermansia muciniphila. Frontiers in Microbiology 8.
5.  Figure 3. Electron microscope images of A. muciniphila. Images by... ResearchGate. https://www.researchgate.net/figure/Electron-microscope-images-of-A-muciniphila-Images-by-Justus-Reunanen-University-of_fig3_283273102. 


---

# Thank you, 
## Any Questions?
```{r echo=FALSE}
happy
```


---
